panel/reactive.py

```diff
@@ -1221,9 +1221,27 @@ def _convert_column(
         dtype = old_values.dtype
         converted: list | np.ndarray | None = None
         if dtype.kind == 'M':
-            if values.dtype.kind in 'if':# TODO: need to doublecheck if this is still needed
-                NATs = np.isnan(values)
-                converted = np.where(NATs, np.nan, values * 10e5).astype(dtype)
+            if values.dtype.kind in 'if':
+                if getattr(dtype, 'tz', None):
+                    # dtype has a timezone
+                    if dtype.tz == dt.timezone.utc:
+                        # Milliseconds to nanoseconds, to datetime64.
+                        converted = (values * 1e6).astype('datetime64[ns]')
+                    else:
+                        import pandas as pd
+
+                        # Using pandas to convert from milliseconds
+                        # timezone-aware, to UTC nanoseconds, to datetime64.
+                        converted = (
+                            pd.Series(pd.to_datetime(values, unit="ms"))
+                            .dt.tz_localize(dtype.tz)
+                            .dt.tz_convert('utc')
+                            .dt.tz_localize(None)
```

Comment:
What is the intention behind these last two lines? This will soon cause pandas to error and also I'm quite confused. This localizes in the original provided timezone, then shifts it to UTC, which seems wrong.

Reply:
I've made a change here https://github.com/holoviz/panel/pull/6889/commits/8d1e9f8dc92696cf98c65408b1233d9c1e20cb00